simulator/scheduler/extender/resultstore/resultstore.go

```diff
@@ -70,74 +67,73 @@ func newData() *result {
 }
 
 // AddStoredResultToPod adds all data corresponding to the specified key to the pod.
-func (s *store) AddStoredResultToPod(pod *v1.Pod) {
+func (s *store) AddStoredResultToPod(pod *v1.Pod) map[string]string {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
 	k := newKey(pod.Namespace, pod.Name)
 	if _, ok := s.results[k]; !ok {
 		// Store doesn't have any scheduling result of the Pod.
-		return
+		return nil
 	}
 
-	if err := s.addFilterResultToPod(pod); err != nil {
+	annotation := map[string]string{}
+	if err := s.addFilterResultToPod(annotation, k); err != nil {
 		klog.Errorf("failed to add filtering result to the pod: %+v", err)
-		return
+		return nil
 	}
 
-	if err := s.addPrioritizeResultToPod(pod); err != nil {
+	if err := s.addPrioritizeResultToPod(annotation, k); err != nil {
 		klog.Errorf("failed to add prioritize result to the pod: %+v", err)
-		return
+		return nil
 	}
 
-	if err := s.addPreemptResultToPod(pod); err != nil {
+	if err := s.addPreemptResultToPod(annotation, k); err != nil {
 		klog.Errorf("failed to add preempt result to the pod: %+v", err)
-		return
+		return nil
 	}
 
-	if err := s.addBindResultToPod(pod); err != nil {
+	if err := s.addBindResultToPod(annotation, k); err != nil {
 		klog.Errorf("failed to add bind result to the pod: $+v", err)
-		return
+		return nil
 	}
+
+	return annotation
 }
 
-func (s *store) addFilterResultToPod(pod *v1.Pod) error {
-	k := newKey(pod.Namespace, pod.Name)
+func (s *store) addFilterResultToPod(anno map[string]string, k key) error {
```

Comment:
It would be better and clearer to rename these methods as well since they are no longer adding a Result to a Pod!
For example ↓
```suggestion
func (s *store) addFilterResultTo(anno map[string]string, k key) error {
func (s *store) addFilterResultToMap(anno map[string]string, k key) error {
```